pkg/fake_pub_server/bin/fake_pub_server.dart

```diff
@@ -25,16 +36,38 @@ Future main(List<String> args) async {
     ].where((e) => e != null).join(' '));
   });
 
+  final datastore = MemDatastore();
   final storage = MemStorage();
 
   final storageServer = FakeStorageServer(storage);
-  final pubServer = FakePubServer(storage);
+  final pubServer = FakePubServer(datastore, storage);
+  final searchService = FakeSearchService(datastore, storage);
+
+  final configuration = Configuration.fakePubServer(
+    frontendPort: port,
+    storageBaseUrl: 'http://localhost:$storagePort',
+    searchPort: searchPort,
+  );
+
+  // On each non-read request, we trigger an update of the search index in
+  // `fake_search_service`.
```

Comment:
Why? Presumably, because you don't want the tests to sleep 30s or so... Then maybe the comment should say so.

But instead of making this so implicit, can't we make an explicit function `await waitForSearchUpdate()` which either:
 * Sleeps 45s when testing against production, or,
 * Calls the `fake-update-all` end-point if testing against the fake setup.

I assume 45s is enough for search to be updated... otherwise make it sleeps longer or shorter time when testing against production deployment.

Reply:
In production the search index update may happen in 10-30 minutes, and we'd need a lot more active index updater process to make it down to 60 seconds or less. I'm not sure that we want to have that before re-thinking how search should work, and I wouldn't aim for such tests to be run against prod.

Reply:
Then maybe the function to update search throws a `UnsupportedError` when running against production..

That way it's more explicit that a test doesn't run against production if it uses this function...

Reply:
Updated. PTAL.